src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/HtmlFormatter.cs

```diff
@@ -53,5 +53,29 @@ internal class HtmlFormatter
 
             return result?.Edits ?? Array.Empty<TextEdit>();
         }
+
+        public async Task<TextEdit[]> FormatOnTypeAsync(
+           FormattingContext context,
+           CancellationToken cancellationToken)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.SourceText.GetLineAndOffset(context.HostDocumentIndex, out var line, out var col);
+            var @params = new DocumentOnTypeFormattingParams()
+            {
+                Position = new Position(line, col),
+                Character = context.TriggerCharacter.ToString(),
```

Comment:
Weird that `DocumentOnTypeFormattingParams.Character` is literally named Character, and yet takes a `string`